src/meta/client/MetaClient.h

```diff
@@ -68,10 +68,18 @@ class MetaClient {
 
     void init();
 
+    void deInit();
+
     void registerListener(MetaChangedListener* listener) {
+        CHECK(listener_ == nullptr);
         listener_ = listener;
     }
 
+    void unRegisterListener() {
+        deInit();
```

Comment:
I understood now. Maybe you should not call deInit here? Because other's still use the bgThread in MetaClient, e.g. SchemaManager?

Reply:
Here we need to stop the background thread first, otherwise it will have race conditions in listener_.

Reply:
We'd better protect listener_ in MetaClient instead of stop bgThread directly.   Because we don't want meta client is controlled by kvstore.

Reply:
According to my understanding.
1 metaclient itself is dependent on the listener_ object.
2 If listener_ is released, both two threads of metaclient will not work, then there is no need for metaclient at this time.

Reply:
Not exactly.  Listener_ is not necessary for meta client.